controllers/quay/quayregistry_controller.go

```diff
@@ -244,10 +244,36 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
+	var userProvidedTLSCert []byte
+	if _, ok := configBundle.Data["ssl.cert"]; ok {
+		err = yaml.Unmarshal(configBundle.Data["ssl.cert"], &userProvidedTLSCert)
+		if err != nil {
+			updatedQuay, err = r.updateWithCondition(&quay, v1.ConditionTypeRolloutBlocked, metav1.ConditionTrue, v1.ConditionReasonConfigInvalid, err.Error())
+			if err != nil {
+				log.Error(err, "failed to update `conditions` of `QuayRegistry`")
+
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	var userProvidedTLSKey []byte
+	if _, ok := configBundle.Data["ssl.key"]; ok {
+		err = yaml.Unmarshal(configBundle.Data["ssl.key"], &userProvidedTLSKey)
+		if err != nil {
+			updatedQuay, err = r.updateWithCondition(&quay, v1.ConditionTypeRolloutBlocked, metav1.ConditionTrue, v1.ConditionReasonConfigInvalid, err.Error())
+			if err != nil {
+				log.Error(err, "failed to update `conditions` of `QuayRegistry`")
+
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	userProvidedCerts := map[string][]byte{"ssl.key": userProvidedTLSKey, "ssl.cert": userProvidedTLSCert}
+
 	updatedQuay.Status.Conditions = v1.RemoveCondition(updatedQuay.Status.Conditions, v1.ConditionTypeRolloutBlocked)
 
 	for _, component := range updatedQuay.Spec.Components {
-		contains, err := kustomize.ContainsComponentConfig(userProvidedConfig, component)
+		contains, err := kustomize.ContainsComponentConfig(userProvidedConfig, userProvidedCerts, component)
```

Comment:
We could pass in `configBundle.Data` directly, no?

Reply:
Not unless we alter how it is called elsewhere as well, since the call to this function in configure.go takes a different shape as configBundle.Data.

Reply:
I don't follow, `configBundle.Data` is of type `map[string][]byte`

Reply:
We can do this, but we will then have to combine reconfigureRequest.Certs and reconfigureRequest.Config into a single map in configure.go

Reply:
I don't follow, sorry. This makes no sense to me. The code "copies" data from A (`configBundle.Data`) to B (`userProvidedCerts`)  if data is present in A, at last it then passes B to the function instead of just passing A and avoid the "copy" all together. Why can't we pass A directly here? As far as I understand nothing needs to be changed in configure.go.